### PR TITLE
Improve the display config

### PR DESCRIPTION
**Pull request by paquiteau**
This PR does two things: 
1. It adds convenient api to update the displayConfig temporarily with a context manager 
2. Add support for custom color cycling by sampling existing matplotlib colormap.

### CODE REVIEW

**Comment by paquiteau** [comment]
Here is a preview of the example file

image

File: src/mrinufft/trajectories/display.py [hunk 1]
```diff
@@ -1,54 +1,121 @@
...
 
-    alpha = 0.2
-    linewidth = 2
-    pointsize = 10
-    fontsize = 18
-    small_fontsize = 14
+    alpha: float = 0.2
+    """Transparency used for area plots, by default ``0.2``."""
+    linewidth: float = 2
+    """Width for lines or curves, by default ``2``."""
```
Comment (Daval-G) [inline src/mrinufft/trajectories/display.py:41]: Is it conventional to comment after the variable definition ?

File: src/mrinufft/trajectories/display.py [hunk 2]
```diff
@@ -1,54 +1,121 @@
...
-    convert_trajectory_to_gradients,
-    KMAX,
     DEFAULT_GMAX,
-    DEFAULT_SMAX,
     DEFAULT_RASTER_TIME,
+    DEFAULT_SMAX,
+    KMAX,
+    compute_gradients_and_slew_rates,
+    convert_trajectory_to_gradients,
 )
```
Comment (Daval-G) [inline src/mrinufft/trajectories/display.py:18]: I like bands

**Review by Daval-G — APPROVED** [review]
LGTM with minor comments.
Great improvement !

File: src/mrinufft/trajectories/display.py (hunk 1 above)
Comment (paquiteau) [reply src/mrinufft/trajectories/display.py:41]: Yes ! It avoids having an "Attributes" section in the docstring. Sphinx then produce a nice looking summary.

**Pull request merged**